pkg/asset/installconfig/ibmcloud/client.go

```diff
@@ -146,6 +166,176 @@ func (c *Client) loadSDKServices() error {
 	return nil
 }
 
+// CreateCOSBucket will create a new COS Bucket in the COS Instance, based on the Instance ID.
+func (c *Client) CreateCOSBucket(ctx context.Context, cosInstanceID string, bucketName string, region string) error {
+	localContext, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+	cosClient := c.getCOSClient(cosInstanceID, region)
+	// Setup Location Constraint, for now we always default to the Smart tier
+	locationConstraint := fmt.Sprintf("%s-%s", region, "smart")
+
+	options := &ibms3.CreateBucketInput{
+		Bucket: ptr.To(bucketName),
+		CreateBucketConfiguration: &ibms3.CreateBucketConfiguration{
+			LocationConstraint: ptr.To(locationConstraint),
+		},
+	}
+	if _, err := cosClient.CreateBucketWithContext(localContext, options); err != nil {
+		return fmt.Errorf("failed to create cos bucket: %w", err)
+	}
+	return nil
+}
+
+// CreateCOSInstance will create a new COS Instance and return the ResourceInstance details.
+func (c *Client) CreateCOSInstance(ctx context.Context, cosName string, resourceGroupID string) (*resourcecontrollerv2.ResourceInstance, error) {
+	localContext, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	createOptions := c.controllerAPI.NewCreateResourceInstanceOptions(cosName, "Global", resourceGroupID, cosServicePlanID)
+
+	instance, _, err := c.controllerAPI.CreateResourceInstanceWithContext(localContext, createOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating new COS instance: %w", err)
+	}
+
+	return instance, nil
+}
+
+// CreateCOSObject will (upload) create a new COS object in the designated COS instance and bucket.
+func (c *Client) CreateCOSObject(ctx context.Context, sourceData []byte, fileName string, cosInstanceID string, bucketName string, region string) error {
+	cosClient := c.getCOSClient(cosInstanceID, region)
+
+	options := &ibms3.PutObjectInput{
+		Body:   cosaws.ReadSeekCloser(bytes.NewReader(sourceData)),
+		Bucket: cosaws.String(bucketName),
+		Key:    cosaws.String(fileName),
+	}
+
+	if _, err := cosClient.PutObject(options); err != nil {
+		return fmt.Errorf("failed creating cos object: %w", err)
+	}
+	return nil
+}
+
+// CreateIAMAuthorizationPolicy creates a new IAM Authorization policy for read access to VPC to a COS Instance.
+func (c *Client) CreateIAMAuthorizationPolicy(ctx context.Context, sourceServiceName string, sourceServiceResourceType string, targetServiceName string, targetServiceInstanceID string, roles []string) error {
+	accountIDKeyPtr := ptr.To("accountId")
+	resourceTypeKeyPtr := ptr.To("resourceType")
+	serviceInstanceKeyPtr := ptr.To("serviceInstance")
+	serviceNameKeyPtr := ptr.To("serviceName")
+	stringEqualsOperatorPtr := ptr.To("stringEquals")
+
+	apiKeyDetails, err := c.GetAuthenticatorAPIKeyDetails(ctx)
+	if err != nil {
+		return fmt.Errorf("failed collecting account ID: %w", err)
+	}
+
+	policyRoles := []iampolicymanagementv1.Roles{}
```

Comment:
Nit: you can pre-allocate this array:
```suggestion
	policyRoles := make([]iampolicymanagementv1.Roles, 0, len(roles))
```

Reply:
Updated.